reference/matrix/csr_kernels.cpp

```diff
@@ -129,9 +130,19 @@ void advanced_spmv(std::shared_ptr<const ReferenceExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_ADVANCED_SPMV_KERNEL);
 
+template <typename Key>
+using unordered_set =
+    std::unordered_set<Key, std::hash<Key>, std::equal_to<Key>,
+                       gko::ExecutorAllocator<Key>>;
+
+template <typename Key, typename Value>
+using unordered_map =
+    std::unordered_map<Key, Value, std::hash<Key>, std::equal_to<Key>,
+                       gko::ExecutorAllocator<std::pair<const Key, Value>>>;
```

Comment:
I think they are general implementation of `unordered_set` and `unordered_map` on `gko::Allocator`, right?
Probably move them into `allocator.hpp`?